### ISSUE TITLE
[Bug Report] [Component] [dropdown, dropdown-item] <el-dropdown> does not meet accessibility requirements

### ISSUE THREAD

**Issue by diana-susanu-nivaura** [element-plus/element-plus#7624]
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.31`
- Element Plus Version: `2.1.8`
- Browser / OS: `Chrome 101.0.4951.54`
- Build Tool: `Webpack`

### Related Component

- `el-dropdown`
- `el-dropdown-item`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuaW1wb3J0IHsgdmVyc2lvbiBhcyBFcFZlcnNpb24gfSBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBBcnJvd0Rvd24gfSBmcm9tICdAZWxlbWVudC1wbHVzL2ljb25zLXZ1ZSdcbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIDxkaXYgY2xhc3M9XCJjb250YWluZXJcIj5cbiAgPGVsLWJ1dHRvbj5DbGljazwvZWwtYnV0dG9uPlxuICAgIDxlbC1kcm9wZG93bj5cbiAgICAgIDxzcGFuIGNsYXNzPVwiZWwtZHJvcGRvd24tbGlua1wiPlxuICAgICAgICBPcHRpb25zIExpc3RcbiAgICAgICAgPGVsLWljb24gY2xhc3M9XCJlbC1pY29uLS1yaWdodFwiPlxuICAgICAgICAgIDxhcnJvdy1kb3duIC8+XG4gICAgICAgIDwvZWwtaWNvbj5cbiAgICAgIDwvc3Bhbj5cbiAgICAgIDx0ZW1wbGF0ZSAjZHJvcGRvd24+XG4gICAgICAgIDxlbC1kcm9wZG93bi1tZW51PlxuICAgICAgICAgIDxlbC1kcm9wZG93bi1pdGVtPk9wdGlvbiAxPC9lbC1kcm9wZG93bi1pdGVtPlxuICAgICAgICAgIDxlbC1kcm9wZG93bi1pdGVtPk9wdGlvbiAyPC9lbC1kcm9wZG93bi1pdGVtPlxuICAgICAgICAgIDxlbC1kcm9wZG93bi1pdGVtPk9wdGlvbiAzPC9lbC1kcm9wZG93bi1pdGVtPlxuICAgICAgICA8L2VsLWRyb3Bkb3duLW1lbnU+XG4gICAgICA8L3RlbXBsYXRlPlxuICAgIDwvZWwtZHJvcGRvd24+XG4gIDwvZGl2PlxuPC90ZW1wbGF0ZT5cblxuPHN0eWxlIHNjb3BlZD5cbi5jb250YWluZXIge1xuICBkaXNwbGF5OiBmbGV4O1xuICBhbGlnbi1pdGVtczogY2VudGVyO1xufVxuLmV4YW1wbGUtc2hvd2Nhc2UgLmVsLWRyb3Bkb3duLWxpbmsge1xuICBjdXJzb3I6IHBvaW50ZXI7XG4gIGNvbG9yOiB2YXIoLS1lbC1jb2xvci1wcmltYXJ5KTtcbiAgZGlzcGxheTogZmxleDtcbiAgYWxpZ24taXRlbXM6IGNlbnRlcjtcbn1cbi5lbC1idXR0b24ge1xuICBtYXJnaW4tcmlnaHQ6IDIwcHg7XG59XG48L3N0eWxlPlxuXG4iLCJpbXBvcnRfbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsIl9vIjp7fX0=)

### Steps to reproduce

Press 'TAB' key. Only button is accessible using TAB key, whereas the dropdown isn't. You cannot open it using keys, which breaks accessibility.

### What is Expected?

Dropdown should be accessible via TAB + Enter keys
And
Oprions inside the dropdown should be accessible using UP + DOWN keys (i.e. you should be able to choose options using arrow keys).
This is bad accessibility, since dropdown is an interactive element.

### What is actually happening?

Dropdown as well as dropdown options aren't accessibile using keyboard navigation.

### Additional comments

_(empty)_

**Comment by Giwayume** [comment]
This has been on my list, working on focus trapping in general right now, which is kind of a prerequisite to fixing dropdown issues.
